### PR TITLE
Define DTTOIF and IFTODT.

### LINKED ISSUE #268: Adding `IFTODT` for more compatibility
Currently `IFTODT` is not defined in wasi-libc, even though `DT_XXX` family are defined.

https://github.com/WebAssembly/wasi-libc/blob/ad5133410f66b93a2381db5b542aad5e0964db96/libc-top-half/musl/include/dirent.h#L50-L65

On the ruby side, we faced a compatibility issue due to missing `IFTODT` https://github.com/ruby/ruby/pull/5614 because we assume `IFTODT` is always defined when `DT_UNKNOWN` is defined, and other platforms are compatible with this.

Do you think it would be worth adding `IFTODT`?

### PR DESCRIPTION
**Pull request by sunfishcode**
Fixes #268.

### CODE REVIEW

File: libc-bottom-half/headers/public/__header_dirent.h [hunk 1]
```diff
@@ -11,6 +11,12 @@
 #define DT_REG __WASI_FILETYPE_REGULAR_FILE
 #define DT_UNKNOWN __WASI_FILETYPE_UNKNOWN
 
+#define IFTODT(x) (__wasilibc_iftodt(x))
+#define DTTOIF(x) (__wasilibc_dttoif(x))
+
+int __wasilibc_iftodt(int x);
+int __wasilibc_dttoif(int x);
```
Comment (sbc100) [inline libc-bottom-half/headers/public/__header_dirent.h:18]: Where are these defined?

Comment (sunfishcode) [reply libc-bottom-half/headers/public/__header_dirent.h:18]: I had forgotten to `git add` these. Fixed now!

**Comment by sunfishcode** [comment]
@sbc100 I forgot to `git add` a file in the original PR here. This is now fixed!

**Review by sbc100 — APPROVED** [review]

**Comment by sunfishcode** [comment]
Thanks!

**Pull request merged**